store/src/store.rs

```diff
@@ -224,119 +272,125 @@ pub trait ChainStore<'a>: Send + Sync + Sized {
-                                }
-                            })
-                    })
-            })
+    fn have_cell(&'a self, out_point: &OutPoint) -> bool {
+        let key = cell_key_from_out_point(out_point);
+        self.get(COLUMN_CELL, &key[..]).is_some()
     }
 
-    fn get_cell_data(
-        &'a self,
-        tx_hash: &packed::Byte32,
-        index: u32,
-    ) -> Option<(Bytes, packed::Byte32)> {
+    fn get_cell_data(&'a self, out_point: &OutPoint) -> Option<(Bytes, packed::Byte32)> {
+        let key = cell_key_from_out_point(out_point);
         if let Some(cache) = self.cache() {
-            if let Some(cached) = cache
-                .cell_data
-                .lock()
-                .get_refresh(&(tx_hash.clone(), index))
-            {
+            if let Some(cached) = cache.cell_data.lock().get_refresh(&key) {
                 return Some(cached.clone());
             }
         };
 
-        let ret = self.get_transaction_info_packed(tx_hash).and_then(|info| {
-            self.get(COLUMN_BLOCK_BODY, info.key().as_slice())
-                .and_then(|slice| {
-                    let reader =
-                        packed::TransactionViewReader::from_slice_should_be_ok(&slice.as_ref());
-                    reader
-                        .data()
-                        .raw()
-                        .outputs_data()
-                        .get(index as usize)
-                        .map(|data| {
-                            (
-                                Unpack::<Bytes>::unpack(&data),
-                                packed::CellOutput::calc_data_hash(&data.raw_data()),
-                            )
-                        })
-                })
+        let ret = self.get(COLUMN_CELL_DATA, &key[..]).map(|slice| {
+            let reader = packed::CellDataEntryReader::from_slice_should_be_ok(&slice.as_ref()[..]);
+            let data = reader.output_data().unpack();
+            let data_hash = reader.output_data_hash().to_entity();
+            (data, data_hash)
         });
 
         if let Some(cache) = self.cache() {
             ret.map(|cached| {
-                cache
-                    .cell_data
-                    .lock()
-                    .insert((tx_hash.clone(), index), cached.clone());
+                cache.cell_data.lock().insert(key, cached.clone());
                 cached
             })
         } else {
             ret
         }
     }
 
+    fn get_transaction_info_packed(
+        &'a self,
+        hash: &packed::Byte32,
+    ) -> Option<packed::TransactionInfo> {
+        self.get(COLUMN_TRANSACTION_INFO, hash.as_slice())
+            .map(|slice| {
+                let reader =
+                    packed::TransactionInfoReader::from_slice_should_be_ok(&slice.as_ref());
+                reader.to_entity()
+            })
+    }
+
+    fn get_transaction_info(&'a self, hash: &packed::Byte32) -> Option<TransactionInfo> {
```

Comment:
What's the difference between `get_transaction_info_packed` and `get_transaction_info`?